src/Plugin/WebformElement/CivicrmSelect.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+namespace Drupal\webform_civicrm\Plugin\WebformElement;
+
+use Drupal\Component\Utility\NestedArray;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Render\Markup;
+use Drupal\webform\Plugin\WebformElementBase;
+use Drupal\webform\WebformSubmissionInterface;
+
+/**
+ * Provides a 'civicrm_select' element.
+ *
+ * @WebformElement(
+ *   id = "civicrm_select",
+ *   label = @Translation("CiviCRM Select"),
+ *   description = @Translation("Provides a CiviCRM powered select list."),
+ *   category = @Translation("CiviCRM"),
+ * )
+ *
+ * @see \Drupal\webform_example_element\Element\WebformExampleElement
+ * @see \Drupal\webform\Plugin\WebformElementBase
+ * @see \Drupal\webform\Plugin\WebformElementInterface
+ * @see \Drupal\webform\Annotation\WebformElement
+ */
+class CivicrmSelect extends WebformElementBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getDefaultProperties() {
+    return parent::getDefaultProperties() + [
+        'form_key' => NULL,
+        'pid' => 0,
+        'value' => '',
+        'empty_option' => '',
+        'empty_value' => '',
+        'options' => [],
+        'options_randomize' => FALSE,
+        'expose_list' => TRUE,
+        'exposed_empty_option' => '- ' . t('Automatic') . ' -',
+        'civicrm_live_options' => 1,
+        'extra' => [
+          'civicrm_live_options' => 1,
+          'items' => [],
+        ],
+      ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function prepare(array &$element, WebformSubmissionInterface $webform_submission = NULL) {
+    parent::prepare($element, $webform_submission);
+    // @todo see how to remove this check, so it has default values.
+    if (empty($element['#options'])) {
+      $exposed = $this->getFieldOptions($element);
+      $element['#options'] = $exposed;
+    }
+  }
+
+  protected function getFieldOptions($element) {
+    $pieces = wf_crm_explode_key($element['#form_key']);
+    list( , , , , $table, $name) = $pieces;
+    $params = ['field' => $name, 'context' => 'create'];
+    return wf_crm_apivalues($table, 'getoptions', $params);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function form(array $form, FormStateInterface $form_state) {
+    $form = parent::form($form, $form_state);
+
+    // Get element properties.
+    $element_properties = $form_state->getValues() ?: $form_state->get('element_properties');
+    $wtf_values = $form_state->getValues();
```

Comment:
`$wtf_values`? :-) I'm assuming that was meant to be `$wf_values` but it's not used anywhere anyway

Reply:
:) Bad debugging statement left behind.